src/satosa/frontends/saml2.py

```diff
@@ -299,12 +300,28 @@ def _handle_authn_response(self, context, internal_response, idp):
             attributes_to_remove = custom_release.get("exclude", [])
             for k in attributes_to_remove:
                 ava.pop(k, None)
-
-        name_id = NameID(text=internal_response.user_id,
-                         format=hash_type_to_saml_name_id_format(
-                             internal_response.user_id_hash_type),
-                         sp_name_qualifier=None,
-                         name_qualifier=None)
+        try:
+            config_nameidformat = self.config['idp_config']['service']['idp']['name_id_format'][0]
```

Comment:
Thanks for overlooking the code vicinity and simplifying it. I kept, however, the try/except block that has the advantage over .get() that a "missing" configuration key can be logged.